clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -846,12 +977,27 @@ Expected<StringRef> wrapSYCLBinariesFromFile(StringRef InputFile,
   M.setTargetTriple(
       Args.getLastArgValue(OPT_host_triple_EQ, sys::getDefaultTargetTriple()));
 
-  StringRef CompileOptions =
+  auto CompileOptionsFromImage =
+      Args.getLastArgValue(OPT_sycl_backend_compile_options_from_image_EQ);
+  auto LinkOptionsFromImage =
+      Args.getLastArgValue(OPT_sycl_backend_link_options_from_image_EQ);
+  auto CompileOptionsFromSYCLBackendCompileOptions =
       Args.getLastArgValue(OPT_sycl_backend_compile_options_EQ);
-  StringRef LinkOptions = Args.getLastArgValue(OPT_sycl_target_link_options_EQ);
+  auto LinkOptionsFromSYCLTargetLinkOptions =
+      Args.getLastArgValue(OPT_sycl_target_link_options_EQ);
+
+  StringRef CompileOptions(
+      Args.MakeArgString(CompileOptionsFromImage.str() +
+                         CompileOptionsFromSYCLBackendCompileOptions.str()));
+  StringRef LinkOptions(Args.MakeArgString(
```

Comment:
This part effects JIT right?

Reply:
Yes. I am not sure if there are any scenarios currently when we might have Device Images inside fat binaries having backend options embedded in them. But, we might end up doing that sometime. 
@mdtoguchi for comment here?

Thanks

Reply:
there is nothing preventing users from using -Xsycl-target* options options to pass along options for either JIT or AOT.  The only expectation is that any values passed by the user be properly formatted for the consuming tool.

Reply:
Yeah sorry I just made the comment to confirm my understanding, looks good. thanks.